File: externals/Async-CTP/samples/VB/Testing/Unit/Testing/Production/Code/SimpleTypes.vb
Imports System.Threading.Tasks

Public Structure SnapshotExtrema
    Public MinDelta As StockSnapshot
    Public MaxDelta As StockSnapshot
End Structure

Public Structure StockSnapshot
    Public Symbol As String
    Public Percent As Double
End Structure

Public Interface IStockSnapshotProvider
    Function GetLatestSnapshotAsync() As Task(Of StockSnapshot)
End Interface

File: externals/Async-CTP/samples/CS/Testing/Unit/Testing/Production/Code/SimpleTypes.cs
using System.Threading.Tasks;

public struct SnapshotExtrema
{
    public StockSnapshot MinDelta;
    public StockSnapshot MaxDelta;
}

public struct StockSnapshot
{
    public string Symbol;
    public double Percent;
}

public interface IStockSnapshotProvider
{
    Task<StockSnapshot> GetLatestSnapshotAsync();
}